lib/view/pages/qr_code_page.dart

```diff
@@ -0,0 +1,131 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:superformula_test/core/resources/extensions.dart';
+import 'package:superformula_test/core/resources/injector.dart';
+import 'package:superformula_test/domain/use_cases/qr_code_get_seed_use_case.dart';
+import 'package:superformula_test/view/blocs/qr_code_bloc/qr_code_bloc.dart';
+import 'package:superformula_test/view/blocs/qr_code_history_bloc/qr_code_history_bloc.dart';
+import 'package:superformula_test/view/widgets/app_loading_widget.dart';
+import 'package:superformula_test/view/widgets/qr_code_error_message.dart';
+import 'package:superformula_test/view/widgets/qr_code_header.dart';
+import 'package:superformula_test/view/widgets/qr_code_history.dart';
+import 'package:superformula_test/view/widgets/qr_code_timer.dart';
+
+class QRCodePage extends StatelessWidget {
+  const QRCodePage({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider<QRCodeCubit>(
+      create: (context) => QRCodeCubit(
+        AppInjector.instance.get<QRCodeGetSeedUseCase>(),
+      ),
+      child: const QRCodeView(),
+    );
+  }
+}
+
+@visibleForTesting
+class QRCodeView extends StatefulWidget {
+  const QRCodeView({super.key});
+
+  @override
+  State<QRCodeView> createState() => QRCodeViewState();
+}
+
+class QRCodeViewState extends State<QRCodeView> {
+  late final QRCodeCubit qrCodeCubit;
+  late final QRCodeHistoryCubit qrCodeHistoryCubit;
+
+  @override
+  void initState() {
+    super.initState();
+    qrCodeCubit = context.read<QRCodeCubit>()..getSeed();
```

Comment:
**question**: the `getSeed()` function is asynchronous and `await`s an HTTP request. Why don't you consume this is in a `FutureBuilder`? What if the server crashed or was slow to respond and the call took a few seconds to execute?

Reply:
The BloC library work with streams. So when I call the `getSeed()` method, I am sending an event through the stream. Once the stream receives that event, it will do all the necessary stuff, while that, my stream sent to my Stream's listeners the data "I am loading" and then "I have succeeded, here is your data!"

That is why I have a lot of BlocListeners and BlocBuilders. I don't need to await for the method because I will be notified of every change of the state by using Streams and it's listeners.